pkg/provider/azure_vmss.go

```diff
@@ -92,18 +96,22 @@ func newScaleSet(az *Cloud) (VMSet, error) {
 	if err != nil {
 		return nil, err
 	}
+	fs, err := newFlexScaleSet(az)
+	if err != nil {
+		return nil, err
+	}
+
 	ss := &ScaleSet{
 		Cloud:           az,
 		availabilitySet: as,
+		flexScaleSet:    fs,
 		vmssVMCache:     &sync.Map{},
 		lockMap:         newLockMap(),
 	}
 
-	if !ss.DisableAvailabilitySetNodes {
-		ss.availabilitySetNodesCache, err = ss.newAvailabilitySetNodesCache()
-		if err != nil {
-			return nil, err
-		}
+	ss.nonVmssUniformNodesCache, err = ss.newNonVmssUniformNodesCache()
```

Comment:
I think we still need the old check: `if !ss.DisableAvailabilitySetNodes || ss.EnableVmssFlexNodes`

Reply:
May I know why do we need the check? Please note the initialization of nonVmssUniformNodesCache only defines the getter function of how to refresh the cache as well as the timeout, but it does not take action to refresh the cache. So, we don't need to worry about the extra api cost.

Reply:
if there are only VMSS in the cluster, the data structure doesn't need to be initialized. This would avoid the related APIs are invoked mistakenly by future changes.

Reply:
Make sense. I just changed the code to add the check !ss.DisableAvailabilitySetNodes || ss.EnableVmssFlexNodes